lib/rest-adapter.js

```diff
@@ -105,6 +104,27 @@ RestAdapter.prototype.connect = function(url) {
   this.connection = url;
 };
 
+/**
+ * Get the auth from the remotes or from the
+ * loopback options if available
+ * @param remotes
+ * @param args
+ * @private
+ */
+RestAdapter.prototype._extractAuth = function(remotes, args) {
+  var auth = remotes.auth;
+  // check for the loopback options to pass along.
+  // It may have the access token that can be used
+  if (auth || !args.length) return auth;
+  var options = args[args.length - 1];
```

Comment:
I am not sure if it's a good idea to always assume the last argument is `options`. Can we perhaps parse remoting metadata for the invoked method to verify that it has `options` argument at the expected position?

Reply:
Done, with a getArgByName in RestMethod.